arch/arm/core/cortex_m/mpu/Kconfig

```diff
@@ -38,3 +38,10 @@ config NXP_MPU
 	default n
 	help
 	  MCU has NXP MPU
+
+config MPU_ALLOW_FLASH_WRITE
+	bool "Add MPU access to write to flash"
+	depends on NXP_MPU
+	default n
```

Comment:
Default options should not make things less secure.  I'd rather the user have to jump through 1 more hoop if they want to specifically allow this.  That's my two cents.